app/controllers/lessons_controller.rb

```diff
@@ -89,13 +89,18 @@ def update
     redirect_to lesson_url(@lesson.id)
   end
 
+  def publish
+    @lesson = Lesson.find(params[:id])
+    @lesson.publish
+ 
+    flash["info"] = "This lesson has been published to Twitter."
+    redirect_to lesson_url(@lesson.id)
+  end
+
   def destroy
```

Comment:
Also not in your work, but does controller action make any sense? I'm not sure why the instance variable would be present frankly, it should be running `@lesson.destroy` after doing a find. Also `destroy` is preferred over delete as it runs any callback and clean up (like deleting tweets if we have that implemented in the model), while `.delete does not`.